src/app/pages/orders/components/orders/orders.component.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  HostBinding,
+  OnInit,
+  inject,
+  DestroyRef,
+} from '@angular/core';
+import { IOrder, OrdersFacadeService, ProductFacadeService } from '@shared-module';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+@Component({
+  selector: 'app-orders',
+  templateUrl: './orders.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OrdersComponent implements OnInit {
+  orders: IOrder[];
+
+  private readonly destroyRef = inject(DestroyRef);
+
+  @HostBinding('class')
+  private readonly classes = 'mx-auto flex items-start flex-wrap py-16 gap-8';
+
+  constructor(
+    private ordersFacadeService: OrdersFacadeService,
+    private productsFacadeService: ProductFacadeService,
+    private cdr: ChangeDetectorRef,
+  ) {}
+
+  async ngOnInit(): Promise<void> {
+    await this.productsFacadeService.initOrdersPage();
+
+    this.ordersFacadeService.getOrderProducts()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((orders: IOrder[]) => {
+        this.orders = orders;
+        this.cdr.detectChanges();
+      });
```

Comment:
maybe let's use async pipe for it? Because we don't need it inside the class as I see.